### DISCUSSION TITLE
How to fix the size of the video default controls

### DISCUSSION THREAD

**Discussion by aaa3334** [vidstack/player#1080] [Q&A]
Hi!
Unsure if its a bug or if I just don't know where to look so posting in here.
Really loving so far the out of the box features here! But am a bit confused still with how everything works. Videos are hosted on s3 with cloudfront.

This is the code I have right now (note I have tried a lot of different things on 'smallLayoutWhen')

const awsurl = "https://df437h3cp7k2g.cloudfront.net/testaws2.mp4";

      <MediaPlayer
        title="Sprite Fight"
        src={awsurl}
        aspectRatio="9/16"
        className="player"
      >
        <DefaultVideoLayout
          icons={defaultLayoutIcons}
          smallLayoutWhen="(width < 5000)"
        />
        <MediaProvider />
      </MediaPlayer>
      

But the layout on the player looks the same every time... If I have a 16:9 video it does not look bad, but the aspect ratio I want is the opposite (mobile videos) so 9:16... and when I do that the controls just dont look good at all (and they are also often not clickable).

I have tried a lot in the css and tried to override things as it says in the docs, but nothing has worked... so looking for someone with more experience to help.

Have added some screenshots below showing the issue - the buttons are just way too large (even if i put it on full screen). Same whether i have 'smallLayoutWhen' in there or not. There is also quite a delay in response from when I click play, but need to sort this first issue out first.

Also happens if I do eg: (removing the 'small player' thing just changes the position in the full screen mode as seen below).

None of this is matching the examples but they are also in a different aspect ratio...

.player {
    width: 180px;
    height:320px

  }

      <MediaPlayer title="Sprite Fight" src={awsurl} className="player">
        <DefaultVideoLayout icons={defaultLayoutIcons} />
        <MediaProvider />
      </MediaPlayer>

Thanks in advance!

<img width="185" alt="Screenshot 2024-01-11 at 12 14 20" src="https://github.com/vidstack/player/assets/63257282/e86dcf26-6fb0-474e-9e43-02b5a3ec82ae">
<img width="175" alt="Screenshot 2024-01-11 at 12 14 34" src="https://github.com/vidstack/player/assets/63257282/e08508e1-204f-41ef-8dec-bef48ee7fad2">
<img width="1470" alt="Screenshot 2024-01-11 at 12 14 44" src="https://github.com/vidstack/player/assets/63257282/e8270d79-fffe-4922-9226-2509b6453e63">
<img width="1470" alt="Screenshot 2024-01-11 at 12 24 21" src="https://github.com/vidstack/player/assets/63257282/9dba28af-8569-400e-a370-54ddd85358d9">

**Comment by mihar-22** [comment]
👋 We haven't tested our Default Layout with vertical videos, you can try setting `smallLayoutWhen` to `true` and seeing if that helps with your needs, otherwise you can consider putting together your own layout that works well for you, we have heaps of [examples](https://github.com/vidstack/examples/tree/main/player/react) showing how to set up.

```tsx
<DefaultVideoLayout smallLayoutWhen />
```


Note: Thanks for trying to provide a detailed question with screenshots but unfortunately none of it helps in figuring out what's happening. It's mostly noise. If you can create a quick repro in the future using our [examples](https://github.com/vidstack/examples/tree/main/player/react) on StackBlitz then I can debug better and see what's up.

**Reply by aaa3334** [reply]
Thanks!
Hm - I will have a go to test it with building my own.. writing like you suggested then did nothing unfortunatly. The examples are quite difficult to follow and know what they are showing is the problem I have with them to be honest - tried to look at them a few times today but haven't managed to get my head around what they are examples of or trying to show past that it is for different libraries. Still trying to figure out how the components go together and what order to place them all in.
The problem is that is I just copied and pasted the code from the setup into a new page in my nextjs14 typescript application (I assume it is compatible with that?)
Will try and build it up from the ground and see if I can get the basic play/pause functionality working without using the DefaultVideoLayout

**Reply by mihar-22** [reply]
You can launch the examples on StackBlitz to see what it looks like live and play around with the code. Find the example you want (e.g., [`react/default-layout`](https://github.com/vidstack/examples/tree/main/player/react/default-layout)), and click the big blue button saying "Open in StackBlitz". You should be able to slowly work your way down from the player component and figure out how it all works.

**Reply by aaa3334** [reply]
Thank you! Taking a look - takes a while for the example to load which was why I didnt realise thats what it was before. Cool feature though!

I figured out what the issue is for anyone else who has had the same issue (maybe super obvious to someone more experienced than me though).
This uses css to style the button component and if you have a global button style (for consistency in an already partially developed app) then that interferes with it.
So I added a local styles.css file, overrode the styling that was the issue and wrapped the video component in a div with className"video_player"

.video_player button {

    width: 30px;
    height: 30px; 
    padding: 0; 
    font-size: 16px; 
  }
